lua/xcodebuild/ui/pickers.lua

```diff
@@ -200,15 +200,22 @@ function M.select_xcodeproj(callback, opts)
   local maxdepth = config.commands.project_search_max_depth
   local sanitizedFiles = {}
   local filenames = {}
-  local files = util.shell(
-    "find '"
+  local cmd = "find '"
+    .. vim.fn.getcwd()
+    .. "' -type d -path '*/.*' -prune -o -maxdepth "
+    .. maxdepth
+    .. " -iname '*.xcodeproj' -print"
+    .. " 2> /dev/null"
+
+  if util.is_fd_installed() then
+    cmd = "fd -I '.*\\.xcodeproj$' '"
       .. vim.fn.getcwd()
-      .. "' -maxdepth "
+      .. "' --max-depth "
       .. maxdepth
-      .. " -iname '*.xcodeproj'"
-      .. " -not -path '*/.*'"
-      .. " 2>/dev/null"
-  )
+      .. " --type d 2> /dev/null"
+  end
+
+  local files = util.shell(cmd)
```

Comment:
The same must be done here in lines 222-223. The pattern must be set based on the approach and the path in `file` must be trimmed.